src/main/java/me/jellysquid/mods/lithium/mixin/world/fast_island_noise/MixinSurfaceChunkGenerator.java

```diff
@@ -0,0 +1,35 @@
+package me.jellysquid.mods.lithium.mixin.world.fast_island_noise;
+
+import me.jellysquid.mods.lithium.common.world.noise.SimplexNoiseCache;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+import net.minecraft.util.math.noise.SimplexNoiseSampler;
+import net.minecraft.world.biome.source.BiomeSource;
+import net.minecraft.world.gen.chunk.ChunkGeneratorType;
+import net.minecraft.world.gen.chunk.SurfaceChunkGenerator;
+
+@Mixin(SurfaceChunkGenerator.class)
+public class MixinSurfaceChunkGenerator {
```

Comment:
This PR claims that it speeds up End generation, but isn't this also implementing the noise cache into the overworld chunk generator?

Reply:
The "surface" chunk generator now functions both for the overworld and the end. I think it was a pretty recent change though